File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        t = 0
        for i in range(len(tickets)):
            if i <= k:
                t = t + min(tickets[i], tickets[k])
            else:
                t = t + min(tickets[i], tickets[k]-1)

        return t

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.kt
class Solution {
    fun timeRequiredToBuy(tickets: IntArray, k: Int): Int {
        var timeTaken = 0
        for (i in tickets.indices) {
            if (i <= k) {
                timeTaken += Math.min(tickets[i], tickets[k])
            } else {
                timeTaken += Math.min(tickets[i], tickets[k]-1)
            }
        }
        return timeTaken;
    }
}